### PR TITLE
Add Android Googlesource Support

### PR DESCRIPTION
**Pull request by mikeyoung85**
https://android.googlesource.com/

### CODE REVIEW

File: lib/android_googlesource_url_parser.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    # but looking at the master branch is the url
+    # https://android.googlesource.com/device/amlogic/yukawa/+/refs/heads/master
+    # and the same applies for tags
+    # https://android.googlesource.com/device/amlogic/yukawa/+/refs/tags/android-12.1.0_r16
+
+    subdir_str = "+"
+    subdir_index = url.index{|part| part == subdir_str}
+    if subdir_index
+      self.url = url[0..subdir_index-1]
+    end
```
Comment (johnbintz-tidelift) [inline lib/android_googlesource_url_parser.rb:42]: Can this whole bit be replaced with a `split` that grabs the first part?

```suggestion
    self.url = url.split('+', 1).first
```

Comment (mikeyoung85) [reply lib/android_googlesource_url_parser.rb:42]: Yes 😄

Comment (mikeyoung85) [reply lib/android_googlesource_url_parser.rb]: Actually it looks like the split method for Arrays comes from Rails and isn't available to the code here...

Comment (mikeyoung85) [reply lib/android_googlesource_url_parser.rb]: This is what I came up with locally to use a `.split('+')` on a String: 
`self.url = url.join("/").split("+").first.chomp("/")` 

I'm not sure if that is much cleaner performance wise, but looks cleaner to read.

**Review by johnbintz-tidelift — APPROVED** [review]

**Pull request merged**